### PR TITLE
DDP-4609 support creating kits with manual approval

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

New requirement is that all the kits created for testboston will need manual approval. Thus, we add `needsApproval` flag to kit configuration so we allow studies to configure this, and add same flag to kit request so DSM will see this flag.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java [hunk 1]
```diff
@@ -10,9 +10,7 @@
     private String participantId;
     private String kitRequestId;
     private String kitType;
-
-    // Flag to satisfy contract with DSM. Should always be set to `false` for now.
-    private boolean needsApproval = false;
+    private boolean needsApproval;
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java:13]: @pegahtah here is the flag that DSM will get.

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Review by MocanaAtBroad — COMMENTED** [review]
Am I understanding correctly that we just mark kits as needing approval and after that it is up to DSM to manage the approval process?

**Comment by yufengwng** [comment]
@MocanaAtBroad yes, Pepper is just marking it. The kit process has always been managed from the DSM side.

**Pull request merged**